### PR TITLE
e2e: add an e2e case BatchResource

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

1. Add an e2e case BatchResource for checking the extended resources is correctly updated when colocation is enabled in the slo-controller-config.
2. Fix the typos of package names.

### Ⅲ. Describe how to verify it

Just run E2E tests.

e.g.

```bash
ginkgo ./test/e2e -- --allowed-not-ready-nodes=5 --system-pods-startup-timeout=10s --e2e-verify-service-account=false
```

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: test/e2e/node/batchresource.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+				for i := range nodeList.Items {
+					node := &nodeList.Items[i]
+					isAllocatable, msg := isNodeHasBatchResourcesAllocatable(node)
+					if isAllocatable {
+						allocatableCount++
+					} else {
+						framework.Logf("node %s has no batch resources allocatable, msg: %s", node.Name, msg)
+					}
+				}
+				if float64(allocatableCount) > float64(totalCount)*minNodesBatchResourceAllocatableRatio {
```
Comment (zwzhang0107) [inline test/e2e/node/batchresource.go:133]: can we also check the exactly batch total resource by mocking NodeMetric CR of node?

File: test/e2e/node/batchresource.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
+/*
```
Comment (zwzhang0107) [inline test/e2e/node/batchresource.go:1]: e2e/node -> e2e/slocontroller ?

**Comment by eahydra** [comment]
NICE! E2E work has finally started!

File: test/e2e/node/OWNERS [hunk 3]
```diff
@@ -0,0 +1,3 @@
+reviewers:
+  - koordlet-approvers
+  - runtime-proxy-approvers
```
Comment (eahydra) [inline test/e2e/node/OWNERS:3]: also need add `approvers` section.
```yaml
approvers:
   - koordlet-approvers
   - runtime-proxy-approvers
```

File: test/e2e/node/batchresource.go (hunk 1 above)
Comment (saintube) [reply test/e2e/node/batchresource.go:133]: Yes, but a little complicated.
By default, the test case runs when the koordinator is deployed in a general configuration. Before mocking a NodeMetric CR, we should disable the node metric reporting by mutating the spec of the koordlet DaemonSet.

Comment (saintube) [reply test/e2e/node/batchresource.go:133]: We may not mock a NodeMetric CR. Instead, we get the latest NodeMetric CR and roughly verify the node allocatable batch resources.

File: test/e2e/node/batchresource.go (hunk 2 above)
Comment (saintube) [reply test/e2e/node/batchresource.go]: done

File: test/e2e/node/OWNERS (hunk 3 above)
Comment (saintube) [reply test/e2e/node/OWNERS]: done. change the approvers as the koordlet's and the slo-controller's

File: test/e2e/node/batchresource.go (hunk 1 above)
Comment (zwzhang0107) [reply test/e2e/node/batchresource.go]: the later one sound good, since NodeMetric won't update frequently.

**Review by jasonliu747 — COMMENTED** [review]
Nice! Should we add a workflow specifically for E2E test?

File: test/e2e/node/batchresource.go (hunk 1 above)
Comment (saintube) [reply test/e2e/node/batchresource.go]: Done. Now we estimate the lower bound of the batch resources using `batch.total >= node.total - node.total * cpuReclaimRatio - nodeMetric.usage - node.total * maxDiffRatio`.

**Comment by saintube** [comment]
> Nice! Should we add a workflow specifically for E2E test?

Yes. Let's add the pipeline later. For now we can use the described command to run locally.

**Review by eahydra — APPROVED** [review]
/approve

**Pull request merged**